clang/lib/AST/ParentMap.cpp

```diff
@@ -34,13 +34,13 @@ static void BuildParentMap(MapTy& M, Stmt* S,
   case Stmt::PseudoObjectExprClass: {
     PseudoObjectExpr *POE = cast<PseudoObjectExpr>(S);
 
-    if (OVMode == OV_Opaque && M[POE->getSyntacticForm()])
+    if (OVMode == OV_Opaque && M.contains(POE->getSyntacticForm()))
```

Comment:
Could put this and the next `if` inside one big `if (M.contains)` check to avoid a duplicated map lookup.

Reply:
Fixed with try_emplace() to avoid the second lookup as well, which inserts when the key is already present.